lib/dry/configurable.rb

```diff
@@ -114,14 +115,30 @@ def _settings
     def _config_for(&block)
       config_klass = ::Class.new { extend ::Dry::Configurable }
       config_klass.instance_eval(&block)
-      config_klass.config
+      create_nested_config(config_klass)
```

Comment:
Perhaps the above two lines can be moved to the new `NestedConfig` class too, an this could just become:

```ruby
::Dry::Configurable::NestedConfig.new(&block)
```